tutorials/nservicebus-101/lesson-1/tutorial.md

```diff
@@ -0,0 +1,144 @@
+
+
+#### Message serializer
+
+snippet:Serializer
+
+When sending messages, an endpoint needs to serialize message objects to a stream, and then deserialize the stream back to a message object on the receiving end. The choice of [**message serializer**](/nservicebus/serialization/) governs what format that will take.
+
+The default serializer is the `XmlSerializer`, which is similar to the built-in .NET XML serializer. XML is the default mostly for historic reasons and to retain backward compatibility. JSON is more compact, more efficient, and easier to integrate with other systems due to its ubiquity. So we will use it instead of the default `XmlSerializer`.
+
+
+#### Persistence
+
+snippet:Persistence
+
+NServiceBus needs [**persistence**](/nservicebus/persistence/) to store some data in between handling messages. We will explore the reasons for this in future lessons but for now, we'll use an implementation that stores everything in memory. This has the advantage during development of allowing us to iterate quickly by providing us with a clean slate every time we start up.
+
+
+#### Error queue
+
+snippet:ErrorQueue
+
+Sometimes processing a message fails due to an unanticipated input, a coding bug, or even something as simple as a database deadlock. Automatic retries will make dealing with deadlocks a non-issue, but for very serious errors, the message could get stuck at the top of the queue and be retried indefinitely. This type of message, known as a **poison message**, would block all other messages behind it. When these occur, NServiceBus needs to be able to set it aside in a different queue to allow other work to get done. This queue is referred to as the **error queue** and is commonly named `error`. We will discuss [**recoverability**](/nservicebus/recoverability/) more in [Lesson 5: Retrying errors](../lesson-5/).
+
+
+#### Installers
+
+snippet:EnableInstallers
+
+This setting instructs the NServiceBus endpoint to run its installers on startup. Installers are used to set up anything the endpoint requires to run. The most common example is creating necessary queues, such as the endpoint's input queue where it will receive messages.
+
+
+### Starting up
+
+At the end of the `AsyncMain` method, after the configuration code, add the following code which will start up the endpoint, keep it running until we press the Enter key, and then shut it down.
+
+snippet:Startup
+
+When the endpoint starts, the `EndpointConfiguration` information is locked down from further changes, and the settings it contains are used to initialize the endpoint.
```

Comment:
Yes it is. It's answering the question "What if I change it after endpoint startup?"